src/NHibernate/Engine/ForeignKeys.cs

```diff
@@ -179,9 +179,16 @@ public static bool IsNotTransientSlow(string entityName, object entity, ISession
 			}
 
 			// let the interceptor inspect the instance to decide
-			// let the persister inspect the instance to decide
-			return session.Interceptor.IsTransient(entity) ??
-			       session.GetEntityPersister(entityName, entity).IsTransient(entity, session);
+			if (session.Interceptor.IsTransient(entity) == true)
+				return true;
+
+			if (entity is INHibernateProxy proxy && proxy.HibernateLazyInitializer.IsUninitialized)
```

Comment:
Can proxy be at all transient? Can't this check be simplified to `enitity is INHibernateProxy`?

Reply:
If I remember correctly it can be when the proxy is used in a different session.

Reply:
I've also realized that user can force entity to be treated as transient by assigning Id to unsaved value. Not sure if it works but let's handle it here

Reply:
I've just noticed the following code:
https://github.com/nhibernate/nhibernate-core/blob/8d13f2492ecd30a1fc451eee38208dbf95a2a614/src/NHibernate/Engine/ForeignKeys.cs#L156-L159

I think we need to use the same logic in both places. But not sure which.

Reply:
Also to decide if this check should be before `interceptor.IsTransient` as in `IsNotTransientSlow`. And should we also add shortcut `session.PersistenceContext.IsEntryFor(entity)`?

`IsNotTransientSlow` should really be implemented as `!IsTransientSlow` otherwise it's doesn't feel right to me that behavior is different.